plugin/trino-memsql/src/test/java/io/trino/plugin/memsql/TestMemSqlTypeMapping.java

```diff
@@ -104,67 +93,62 @@ public final void destroy()
     @Test
     public void testBasicTypes()
     {
-        DataTypeTest.create()
-                .addRoundTrip(bigintDataType(), 123_456_789_012L)
-                .addRoundTrip(integerDataType(), 1_234_567_890)
-                .addRoundTrip(smallintDataType(), (short) 32_456)
-                .addRoundTrip(tinyintDataType(), (byte) 125)
-                .addRoundTrip(doubleDataType(), 123.45d)
-                .addRoundTrip(realDataType(), 123.45f)
+        SqlDataTypeTest.create()
+                .addRoundTrip("bigint", "123456789012", BIGINT, "123456789012")
+                .addRoundTrip("integer", "1234567890", INTEGER, "1234567890")
+                .addRoundTrip("smallint", "32456", SMALLINT, "SMALLINT '32456'")
+                .addRoundTrip("tinyint", "125", TINYINT, "TINYINT '125'")
+                .addRoundTrip("double", "123.45", DOUBLE, "DOUBLE '123.45'")
+                // TODO: fails in SqlDataTypeTest#verifyPredicate
+                // .addRoundTrip("real", "123.45", REAL, "REAL '123.45'")
                 .execute(getQueryRunner(), trinoCreateAsSelect("test_basic_types"));
     }
 
     @Test
     public void testFloat()
     {
-        singlePrecisionFloatingPointTests(realDataType())
-                .execute(getQueryRunner(), trinoCreateAsSelect("trino_test_float"));
-        singlePrecisionFloatingPointTests(memSqlFloatDataType())
-                .execute(getQueryRunner(), memSqlCreateAndInsert("tpch.memsql_test_float"));
-    }
+        // TODO: convert to SqlDataTypeTest (fails in SqlDataTypeTest#verifyPredicate)
 
-    private static DataTypeTest singlePrecisionFloatingPointTests(DataType<Float> floatType)
-    {
         // we are not testing Nan/-Infinity/+Infinity as those are not supported by MemSQL
-        return DataTypeTest.create()
-                .addRoundTrip(floatType, 3.14f)
+        DataTypeTest.create()
+                .addRoundTrip(realDataType(), 3.14f)
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [CNT: Rough value of known constant found (CNT_ROUGH_CONSTANT_VALUE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#cnt-rough-value-of-known-constant-found-cnt-rough-constant-value).
It's recommended to use the predefined library constant for code clarity and better precision.

Reply:
What constant would you recommend here instead?

Reply:
`Math.PI` instead for getting better precision. :)

Reply:
Right, that would be helpful if we wanted to test with PI number.
The point of this test is not that, however. It just uses some number (with intentionally small decimal precision), to exercise real (float) read/write path.

Reply:
... And it uses something that looks like π just because someone found this amusing. 
We use `42` for the same reason quite often.

Hope that explains.

Reply:
Noticed. Thanks a lot. :)